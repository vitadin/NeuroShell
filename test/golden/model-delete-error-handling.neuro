%% Test model deletion error handling
\set[_echo_command="true"]

%% Create models for testing error scenarios
\model-new[catalog_id="O4M"] test-model-1
\model-new[catalog_id="O3"] test-model-2

%% Test no matches scenario
\try \model-delete nonexistent-model

%% Test multiple matches scenario
\model-new[catalog_id="O4M"] test-similar
\model-new[catalog_id="G41"] test-different
\try \model-delete test

%% Test empty input error
\try \model-delete 

%% Test non-existent ID prefix
\try \model-delete[id=true] xyz123

%% Clean up for final test
\model-delete test-model-1
\model-delete test-model-2  
\model-delete test-similar
\model-delete test-different

%% Test when no models exist at all
\try \model-delete anything