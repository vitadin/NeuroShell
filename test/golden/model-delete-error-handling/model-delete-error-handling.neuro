%% Test model deletion error handling
\set[_echo_command="true"]

%% Create models for testing error scenarios
\model-new[catalog_id="CS4"] test-model-1
\model-new[catalog_id="O3"] test-model-2

%% Test no matches scenario
\try \model-delete nonexistent-model

%% Test multiple matches scenario
\model-new[provider="anthropic", base_model="claude-3-sonnet"] test-similar
\model-new[provider="openai", base_model="gpt-4"] test-different
\try \model-delete test

%% Test empty input error
\try \model-delete 

%% Test non-existent ID prefix
\try \model-delete[id=true] xyz123

%% Clean up for final test
\model-delete test-model-1
\model-delete test-model-2  
\model-delete test-similar
\model-delete test-different

%% Test when no models exist at all
\try \model-delete anything