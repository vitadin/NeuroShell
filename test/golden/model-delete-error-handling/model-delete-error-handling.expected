Setting _echo_command = true
%%> "\\model-new[catalog_id=\"CS4\"] test-model-1"
Created model 'test-model-1' (ID: 00000001, Provider: anthropic, Base: claude-sonnet-4-20250514)
%%> "\\model-new[catalog_id=\"O3\"] test-model-2"
Created model 'test-model-2' (ID: 00000002, Provider: openai, Base: o3)
%%> "\\try \\model-delete nonexistent-model"
%%> "\\model-delete nonexistent-model"
%%> "\\model-new[provider=\"anthropic\", base_model=\"claude-3-sonnet\"] test-similar"
Created model 'test-similar' (ID: 00000003, Provider: anthropic, Base: claude-3-sonnet)
%%> "\\model-new[provider=\"openai\", base_model=\"gpt-4\"] test-different"
Created model 'test-different' (ID: 00000004, Provider: openai, Base: gpt-4)
%%> "\\try \\model-delete test"
%%> "\\model-delete test"
%%> "\\try \\model-delete"
%%> "\\model-delete"
%%> "\\try \\model-delete[id=true] xyz123"
%%> "\\model-delete[id=true] xyz123"
%%> "\\model-delete test-model-1"
Deleted model 'test-model-1' (ID: 00000001, Provider: anthropic, Base: claude-sonnet-4-20250514)
%%> "\\model-delete test-model-2"
Deleted model 'test-model-2' (ID: 00000002, Provider: openai, Base: o3)
%%> "\\model-delete test-similar"
Deleted model 'test-similar' (ID: 00000003, Provider: anthropic, Base: claude-3-sonnet)
%%> "\\model-delete test-different"
Deleted model 'test-different' (ID: 00000004, Provider: openai, Base: gpt-4)
%%> "\\try \\model-delete anything"
%%> "\\model-delete anything"